### ISSUE TITLE
Formula installer can't find suitable pytest-runner

### ISSUE THREAD

**Issue by sursh** [iterative/homebrew-dvc#8]
When installing through Homebrew and the formula you've provided, I get this error: 

```$ brew install iterative/homebrew-dvc/dvc
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
azure-cli                                                                       fn                                                                              libcouchbase

==> Installing dvc from iterative/dvc
==> Downloading https://github.com/iterative/dvc/archive/0.19.0.tar.gz
Already downloaded: /Users/sasha/Library/Caches/Homebrew/downloads/b5fdfdb3fadc651a8d95fb8a3f7b92256b693e1611e30615a8e2359f72548a91--dvc-0.19.0.tar.gz
Warning: Cannot verify integrity of b5fdfdb3fadc651a8d95fb8a3f7b92256b693e1611e30615a8e2359f72548a91--dvc-0.19.0.tar.gz
A checksum was not provided for this resource
For your reference the SHA256 is: 4b4816cf23293835f4d7f0cfc063f6a69e5fc0d3a021254d896650d297ba732e
==> Downloading https://files.pythonhosted.org/packages/33/bc/fa0b5347139cd9564f0d44ebd2b147ac97c36b2403943dbee8a25fd74012/virtualenv-16.0.0.tar.gz
Already downloaded: /Users/sasha/Library/Caches/Homebrew/downloads/02e9f856894cc3f1979f2bf31e2f709c30ef64ea80ce14a80cc83e384528d06f--virtualenv-16.0.0.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/private/tmp/dvc--homebrew-virtualenv-20180921-11323-1lv5w5h/target --single-version-externally-managed --record=installed.txt
==> python -s /private/tmp/dvc--homebrew-virtualenv-20180921-11323-1lv5w5h/target/bin/virtualenv -p python /usr/local/Cellar/dvc/0.19.0/libexec
==> /usr/local/Cellar/dvc/0.19.0/libexec/bin/pip install -r requirements.txt
Last 15 lines from /Users/sasha/Library/Logs/Homebrew/dvc/03.pip:
        replace_conflicting=True,
      File "/usr/local/Cellar/dvc/0.19.0/libexec/lib/python2.7/site-packages/pkg_resources/__init__.py", line 777, in resolve
        replace_conflicting=replace_conflicting
      File "/usr/local/Cellar/dvc/0.19.0/libexec/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1060, in best_match
        return self.obtain(req, installer)
      File "/usr/local/Cellar/dvc/0.19.0/libexec/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1072, in obtain
        return installer(requirement)
      File "/usr/local/Cellar/dvc/0.19.0/libexec/lib/python2.7/site-packages/setuptools/dist.py", line 581, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/local/Cellar/dvc/0.19.0/libexec/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 670, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pytest-runner')

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/tmp/pip-install-bG526S/reflink/
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `write': Broken pipe (Errno::EPIPE)
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `puts'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `rescue in block (3 levels) in safe_fork'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:31:in `block (3 levels) in safe_fork'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `fork'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `block (2 levels) in safe_fork'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `open'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `block in safe_fork'
        from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/tmpdir.rb:89:in `mktmpdir'
        from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:26:in `safe_fork'
        from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:719:in `build'
        from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:311:in `install'
        from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:321:in `install_formula'
        from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:253:in `block in install'
        from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:251:in `each'
        from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:251:in `install'
        from /usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/iterative/homebrew-dvc/issues
```

I've tried installing pytest-runner globally before proceeding with `pip2 install pytest-runner` (which installs v4.2) but I still get the same error. Any ideas?

**Comment by efiop** [comment]
Hi @sursh !

Hm, 

> I've tried installing pytest-runner globally before proceeding with pip2 install pytest-runner (which installs v4.2) but I still get the same error.

That is because brew uses its own python to build the package with. And `pip2 install` installs to system python(or Anaconda's, depending on your environment), thus it doesn't affect the brew installation.

Looks like it is simply a missing dependency in our requirements, though, strangely enough, I can't reproduce it on High Sierra on my machine nor on Travis, so maybe it is also somehow caused by the older OS X version. Investigating this right now.

Thanks,
Ruslan

**Comment by efiop** [comment]
Hi @sursh !

I've updated some requirements for dvc, added tests for formulas on older OSX versions(which are passing successfully) and released 0.19.2. Could you please try installing from homebrew formula once again to see if it works now?

Thanks,
Ruslan

**Comment by sursh** [comment]
Thank you for the extremely quick and thorough responses to both of my issues! I've confirmed that the formula installation works now. Excited to use dvc!
